egs/wsj/s5/local/nnet3/run_discriminative.sh

```diff
@@ -0,0 +1,224 @@
+    # the phases of get_egs_discriminative2.sh below.
+    num_threads_denlats=6
+    subsplit=40 # number of jobs that run per job (but 2 run at a time, so total jobs is 80, giving
+    # total slots = 80 * 6 = 480.
+    steps/nnet3/make_denlats.sh --cmd "$decode_cmd --mem 1G --num-threads $num_threads_denlats" \
+      --online-ivector-dir $online_ivector_dir \
+      --nj $nj --sub-split $subsplit --num-threads "$num_threads_denlats" --config conf/decode_dnn.config \
+      $train_data_dir data/lang $srcdir $lats_dir || exit 1;
+  fi
+fi
+
+if [ $stage -le 2 ]; then
+  # hardcode no-GPU for alignment, although you could use GPU [you wouldn't
+  # get excellent GPU utilization though.]
+  nj=100 # have a high number of jobs because this could take a while, and we might
+         # have some stragglers.
+  use_gpu=no
+  gpu_opts=
+
+  steps/nnet3/align.sh  --cmd "$decode_cmd $gpu_opts" --use-gpu "$use_gpu" \
+     --online-ivector-dir $online_ivector_dir \
+     --nj $nj $train_data_dir data/lang $srcdir ${srcdir}_ali || exit 1;
+
+  # the command below is a more generic, but slower, way to do it.
+  # steps/online/nnet2/align.sh --cmd "$decode_cmd $gpu_opts" --use-gpu "$use_gpu" \
+  #    --nj $nj data/train_960 data/lang ${srcdir}_online ${srcdir}_ali || exit 1;
+fi
+
+left_context=14
+right_context=10
+
+cmvn_opts=`cat $srcdir/cmvn_opts` || exit 1
+
+if [ -z "$degs_dir" ]; then
+  degs_dir=${srcdir}_degs_n${frames_per_eg}_o${frames_overlap_per_eg}_f
+  if $determinize; then
+    degs_dir=${degs_dir}d
+  fi
+  if $minimize; then
+    degs_dir=${degs_dir}m
+  fi
+  if $remove_output_symbols; then
+    degs_dir=${degs_dir}r
+  fi
+  if $remove_epsilons; then
+    degs_dir=${degs_dir}e
+  fi
+  if $collapse_transition_ids; then
+    degs_dir=${degs_dir}c
+  fi
+
+  if [ $stage -le 3 ]; then
+    if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d ${srcdir}_degs/storage ]; then
+      utils/create_split_dir.pl \
+        /export/b0{1,2,5,6}/$USER/kaldi-data/egs/swbd-$(date +'%m_%d_%H_%M')/s5/${srcdir}_degs/storage ${srcdir}_degs/storage
+    fi
+    # have a higher maximum num-jobs if
+    if [ -d ${srcdir}_degs/storage ]; then max_jobs=10; else max_jobs=5; fi
+
+    degs_opts="--determinize $determinize --minimize $minimize --remove-output-symbols $remove_output_symbols --remove-epsilons $remove_epsilons --collapse-transition-ids $collapse_transition_ids"
+
+    steps/nnet3/get_egs_discriminative.sh \
+      --cmd "$decode_cmd --max-jobs-run $max_jobs --mem 20G" --stage $get_egs_stage --cmvn-opts "$cmvn_opts" \
+      --online-ivector-dir $online_ivector_dir --left-context $left_context --right-context $right_context \
+      --criterion $criterion --frames-per-eg $frames_per_eg --frames-overlap-per-eg $frames_overlap_per_eg ${degs_opts} \
+      $train_data_dir data/lang ${srcdir}_ali $lats_dir $srcdir/final.mdl $degs_dir || exit 1;
+
+    # the command below is a more generic, but slower, way to do it.
```

Comment:
vimal, you did not act on this comment.
It looks like you have seen only a minority of my comments.